src/System.Private.CoreLib/shared/System/Collections/Generic/IComparer.cs

```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```

Comment:
Compare should have a // TODO-NULLABLE-GENERIC comment on it.  Implementations need to work with nulls.  According to the docs: "Comparing null with any reference type is allowed and does not generate an exception."